File: cs50x-2022/week1/mario/more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    
    // User chooses pyramid height 
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    
    // Lines
    for (short i = 0; i < height; i++)
    {
        // Left side spaces
        for (short j = 0; j < (height - 1) - i; j++)
        {
            printf(" "); 
        }
        
        // Left side hashes
        for (short j = 0; j <= i; j++)
        {
            printf("#"); 
        }
        
        // Middle spaces 
        printf("  ");
        
        // Right side
        for (short j = 0; j <= i; j++)
        {
            printf("#"); 
        }
        
        // Line feed
        printf("\n");
    }
}

File: cs50x-2022/week6/sentimental-mario/more/mario.py
from cs50 import get_int


def main():
    # checks the input
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    # builds the pyramid
    for i in range(height):
        print(" " * (height - 1 - i), end="")
        print("#" * (i + 1), end="")
        print("  ", end="")
        print("#" * (i + 1))


if __name__ == "__main__":
    main()